nchelpers/__init__.py

```diff
@@ -0,0 +1,405 @@
+            'lev': 'Z',
+            'level': 'Z'
+        }
+        return {dim_to_axis[dim]: dim for dim in dim_names if dim in dim_to_axis}
+
+    def dim_axes(self, dim_names=None):
+        """Return a dictionary mapping specified dimension names (or all dimensions in file) to
+        the canonical axis name for each dimension.
+        
+        :param dim_names: (str) List of names of dimensions of interest, None for all dimensions in file
+        :return: (dict) Dictionary mapping dimension name to canonical axis name, for specified dimension names
+        """
+        if not dim_names:
+            dim_names = self.dim_names()
+
+        if len(dim_names) == 0:
+            return {}
+
+        # Start with our best guess
+        axis_to_dim = self.dim_axes_from_names(dim_names)
+
+        # Then fill in the rest from the 'axis' attributes
+        # TODO: Does this happen? i.e., when are dimension names the same as axis names?
+        # Alternatively, is this some kind of (relatively benign) programming error?
+        for axis in axis_to_dim.keys():
+            if axis in self.dimensions and axis in self.variables \
+                    and hasattr(self.variables[axis], 'axis'):
+                axis_to_dim[axis] = self.variables[axis].axis
+
+                # Apparently this is how a "space" dimension is attributed?
+                if hasattr(self.variables[axis], 'compress'):
+                    axis_to_dim[axis] = 'S'
+
+        # Invert {axis: dim} to {dim: axis}
+        return {dim: axis for axis, dim in axis_to_dim.items()}
+
+    @property
+    def climatology_bounds_var_name(self):
+        axes = self.dim_axes()
+        if 'T' in axes:
+            time_axis = axes['T']
+        else:
+            return None
+
+        # TODO: Do we really mean 'climatology' in self.variables[time_axis].ncattrs()? If so, use that. This looks
+        # imprecise and hard to understand
+        if 'climatology' in self.variables[time_axis]:
+            return self.variables[time_axis].climatology
+        else:
+            return None
+
+    @property
+    def is_multi_year_mean(self):
+        """True if the metadata indicates that the data consists of a multi-year mean"""
+        # TODO: Is it really true that every data file that consists of a multi-year mean actually (should) contain a
+        # time dimension with attribute 'climatology'? Or is there a better condition, perhaps based on cell_method?
+        return bool(self.climatology_bounds_var_name)
+
+    @property
+    def time_var(self):
+        """The time variable (netCDF4.Variable) in this file"""
+        axes = self.dim_axes_from_names()
+        if 'T' in axes:
+            time_axis = axes['T']
+        else:
+            raise ValueError("No axis is attributed with time information")
+        t = self.variables[time_axis]
+        assert hasattr(t, 'units') and hasattr(t, 'calendar')
+        return t
+
+    @property
+    def time_steps(self):
```

Comment:
Problem with lru_cache, which I looked earlier, is that it is Py3 only. There is a backport for Py2 but that introduces ugliness because it is *only* for Py2 and can't be installed in Py3. Sigh.

There is a likely nicer solution at https://pypi.python.org/pypi/cached-property